backend/server/routers/programs.py

```diff
@@ -136,22 +136,28 @@ def getMinors(programCode: str):
     return {"minors": minrs}
 
 def convertSubgroupObjectToCoursesDict(object: str, description: str|list[str]) -> dict[str, str]:
-    """ gets a subgroup object (format layed out in the processor) and fetches the exact courses its referring to """
+    """ Gets a subgroup object (format laid out in the processor) and fetches the exact courses its referring to """
     if " or " in object:
         return {c: description[index] for index, c in enumerate(object.split(" or "))}
     elif not re.match(r"[A-Z]{4}[0-9]{4}", object):
-        return search(object)
+        return regex_search(object)
     else:
         return {object: description}
 
 def addSubgroupContainer(structure: dict, type: str, container: dict, exceptions: list[str]) -> list[str]:
-    """returns the added courses"""
+    """ Returns the added courses """
     structure[type][container["title"]] = {}
     item = structure[type][container["title"]]
 
     item["UOC"] = container["credits_to_complete"]
     item["courses"] = {}
     for object, description in container["courses"].items():
+        if " or " in object:
+            courses_mentioned = {c: description[index] for index, c in enumerate(object.split(" or "))}
```

Comment:
Where is this variable used?

Reply:
bro i swear i never wrote this part of the code it just appeared there lmao

Reply:
but ur right this looks like useless code

Reply:
literally have no idea how it got there